samcli/commands/check/resources/template_resource.py

```diff
@@ -0,0 +1,37 @@
+"""
+Super class for all resources
+"""
+import abc
+
+from typing import Any
+from samcli.lib.providers.provider import Function
+
+
+class TemplateResource:
+    _resource_object: Function
+    _resource_type: str
+
+    def __init__(self, resource_object: Function, resource_type: str):
+        """
+        Parameters
+        ----------
+            resource_object: Function
+                The resource object form the template file
+            resource_type: str
+                The resource type
+        """
+        self._resource_object: Function = resource_object
+        self._resource_type: str = resource_type
+
+    @property
+    def resource_object(self) -> Function:
+        return self._resource_object
+
+    @property
+    @abc.abstractmethod
+    def resource_name(self) -> Any:
+        raise NotImplementedError()
```

Comment:
Just curious, why we have `resource_name` as abstract method? I assume this is just a string property, we may think about assigning it in the constructor rather than making it abstract.